scipy/sparse/tests/test_base.py

```diff
@@ -59,6 +59,23 @@ def assert_array_equal_dtype(x, y, **kwargs):
     assert_(x.dtype == y.dtype)
     assert_array_equal(x, y, **kwargs)
 
+def _underlying_arrays(x):
```

Comment:
This is definitely a bit of magic/hackery. It may be safer to enumerate the underlying array names for the various formats, as it shouldn't be too many distinct cases.

Either way, a comment explaining what this is used for and why it's here would be good.

Reply:
@perimosocordiae It would be `[indices, indptr, data, offsets, row, col]`, but we would still need to check if they are array instances and it would be more prone to miss one on future changes. Isn't it better to catch all attributes with array types?